starknet-client/src/client/chain_state.rs

```diff
@@ -105,25 +107,24 @@ impl StarknetClient {
     pub async fn poll_transaction_completed(
         &self,
         tx_hash: StarknetFieldElement,
-    ) -> Result<TransactionReceipt, StarknetClientError> {
+    ) -> Result<TransactionStatus, StarknetClientError> {
+        // On devnet, we do not generally have access to the pathfinder API, and transaction
+        // finality is essentially instant, so we skip the polling
+        match self.config.chain {
```

Comment:
this looks fine to me, but can we add a debug log saying that polling is being skipped on devnet?